packages/connectors/src/metaMask.ts

```diff
@@ -31,10 +31,27 @@ import {
   withTimeout,
 } from 'viem'
 
-export type MetaMaskParameters = Compute<
-  ExactPartial<Omit<MetaMaskSDKOptions, '_source' | 'readonlyRPCMap'>>
+type WagmiMetaMaskSDKOptions = Compute<
+  ExactPartial<
+    Omit<
+      MetaMaskSDKOptions,
+      | '_source'
+      | 'readonlyRPCMap'
+      | 'useDeeplink'
+      | 'injectProvider'
+      | 'forceInjectProvider'
+      | 'forceDeleteProvider'
+    >
+  >
 >
 
+export type MetaMaskParameters = WagmiMetaMaskSDKOptions & {
+  // Shortcut to connect and sign a message
+  connectAndSign?: string
+  // Allow connectWith any rpc method
+  connectWith?: { method: string; params: unknown[] }
```

Comment:
```suggestion
  connectWith?: { method: string; params: unknown[] } | undefined
```

What `method`s are supported? We should be able to strongly type this with the utility types in Viem (e.g. `EIP1474Methods`) if we want.